src/pages/user/membership.tsx

```diff
@@ -0,0 +1,186 @@
+import { shortenPlanInterval } from '~/components/Stripe/stripe.utils';
+import { EdgeMedia } from '~/components/EdgeMedia/EdgeMedia';
+import { NextLink } from '@mantine/next';
+import { trpc } from '~/utils/trpc';
+import { getPlanDetails } from '~/components/Stripe/PlanCard';
+import { useFeatureFlags } from '~/providers/FeatureFlagsProvider';
+import { PlanBenefitList } from '~/components/Stripe/PlanBenefitList';
+import { IconDotsVertical, IconRotateClockwise } from '@tabler/icons-react';
+import { ProductMetadata } from '~/server/schema/stripe.schema';
+import { constants } from '~/server/common/constants';
+import { dialogStore } from '~/components/Dialog/dialogStore';
+import { CancelMembershipFeedbackModal } from '~/components/Stripe/MembershipChangePrevention';
+import { SubscribeButton } from '~/components/Stripe/SubscribeButton';
+
+export const getServerSideProps = createServerSideProps({
+  useSession: true,
+  resolver: async ({ session, ctx }) => {
+    if (!session || !session.user)
+      return {
+        redirect: {
+          destination: getLoginLink({ returnUrl: ctx.resolvedUrl }),
+          permanent: false,
+        },
+      };
+
+    if (!session.user.subscriptionId)
+      return {
+        redirect: {
+          destination: '/pricing',
+          permanent: false,
+        },
+      };
+  },
+});
+
+const useStyles = createStyles((theme) => ({
+  card: {
+    height: '100%',
+    background: theme.colorScheme === 'dark' ? theme.colors.dark[8] : theme.colors.gray[0],
+    borderRadius: theme.radius.md,
+    padding: theme.spacing.lg,
+  },
+  price: {
+    fontSize: 48,
+    fontWeight: 700,
+  },
+}));
+
+export default function UserMembership() {
+  const { classes, theme } = useStyles();
+  const { data: subscription, isLoading } = trpc.stripe.getUserSubscription.useQuery();
+  const features = useFeatureFlags();
+
+  if (isLoading || !subscription) {
+    return (
+      <Container size="lg">
+        <Center>
+          <Loader />
+        </Center>
+      </Container>
+    );
+  }
+
+  const price = subscription.price;
+  const product = subscription.product;
+  const { image, benefits } = getPlanDetails(subscription.product, features);
+  const productMeta = (product.metadata ?? {}) as ProductMetadata;
+  const canUpgrade =
+    productMeta.tier !==
+    constants.memberships.tierOrder[constants.memberships.tierOrder.length - 1];
+
+  return (
+    <>
+      <Meta title="My Membership" deIndex="noindex, nofollow" />
+      <Container size="md">
+        <Grid>
+          <Grid.Col span={12}>
+            <Stack>
+              <Title>My Membership Plan</Title>
+              <Paper withBorder className={classes.card}>
+                <Stack>
+                  <Group position="apart">
+                    <Group>
+                      {image && (
+                        <Center>
+                          <EdgeMedia src={image} width={100} />
```

Comment:
Would it be better to keep the width consistent because of cache? You're using the same thing here https://github.com/civitai/civitai/pull/1084/files#diff-0a07a6e417dc8e6d1634c145869bfcdce49be581818e95e40629eb0eb819919fR56
```suggestion
                          <EdgeMedia src={image} width={40} />
```